utils/utils.go

```diff
@@ -41,6 +45,33 @@ func GetInstancesInSecurityGroup(ctx context.Context, client *egoscale.Client, s
 	return instancesInSG, nil
 }
 
+func GetInstancesAttachedToEIP(ctx context.Context, client *egoscale.Client, elasticIPID, zone string) ([]*v2.Instance, error) {
+	allInstances, err := client.ListInstances(ctx, zone)
+	if err != nil {
+		return nil, err
+	}
+
+	var instancesAttachedToEIP []*v2.Instance
+	for _, instance := range allInstances {
+		completeInstanceInfo, err := client.GetInstance(ctx, zone, *instance.ID)
```

Comment:
I'm not sure what to do here. ListInstance returns a slice of Instances which has the field ElasticPIIDs but it's nil even when the instances have an elastic IP. So to get them, I do a GetInstance call here on each instance.
I thought this was a bug, but when I researched it I realized that ListInstances was removed in newer versions of egoscale.

Another option I found was https://github.com/exoscale/cli/blob/master/cmd/eip_show.go#L116
But this command is deprecated.

What's the best way to list instances attached to an EIP?

Reply:
TBH, I'm not sure there is "a best way" :)